src/arviz_stats/base/core.py

```diff
@@ -252,40 +252,94 @@ def _hdi_nearest(self, ary, prob, circular, skipna):
 
         return hdi_interval
 
-    def _hdi_multimodal(self, ary, prob, skipna, max_modes):
+    def _hdi_multimodal_continuous(self, ary, prob, skipna, max_modes, circular, **kwargs):
         """Compute HDI if the distribution is multimodal."""
         ary = ary.flatten()
         if skipna:
             ary = ary[~np.isnan(ary)]
 
-        if ary.dtype.kind == "f":
-            bins, density, _ = self.kde(ary)
-            lower, upper = bins[0], bins[-1]
-            range_x = upper - lower
-            dx = range_x / len(density)
+        bins, density, _ = self.kde(ary, circular=circular, **kwargs)
+        lower, upper = bins[0], bins[-1]
+        range_x = upper - lower
+        dx = range_x / len(density)
+
+        hdi_intervals, interval_probs = self._hdi_from_bin_probabilities(
+            bins, density, prob, circular, dx
+        )
+
+        return self._pad_hdi_to_maxmodes(hdi_intervals, interval_probs, max_modes)
+
+    def _hdi_multimodal_discrete(self, ary, prob, bins, max_modes):
+        """Compute HDI if the distribution is multimodal."""
+        ary = ary.flatten()
+
+        if bins is None:
+            bins, density = np.unique(ary, return_counts=True)
+            dx = 1
         else:
             bins = self._get_bins(ary)
-            density, _ = self._histogram(ary, bins=bins, density=True)
+            density, _ = self._histogram(ary, bins=bins)
```

Comment:
This should probably be `density, bins = self._histogram(ary, bins=bins)` instead. Similary to `experimental` being somewhat nice for plots but not a good choice for multimodal kde, I think `_get_bins` defaults and assumptions are probably not good for multimodal hdi either.